src/org/traccar/BaseProtocolDecoder.java

```diff
@@ -31,6 +31,22 @@
 
     private final Protocol protocol;
 
+    public long addUnknownDevice(String uniqueId, SocketAddress remoteAddress) {
+        long deviceId = 0;
+        Device newDevice = new Device();
+        newDevice.setName(uniqueId);
+        newDevice.setUniqueId(uniqueId);
+
+        try {
+            Context.getDeviceManager().addDevice(newDevice);
+            Log.info("Nice meeting new device " + uniqueId + ".");
```

Comment:
No problem.  I'm trying to stick to your standards.  (Though personally I'm all for friendly messages.  I like machine with a sense of humor, for example.  But I'm not a person who insists on any style of anything.  What I can't tolerate is software with flaws.)